kong/plugins/prometheus/enterprise/exporter.lua

```diff
@@ -0,0 +1,95 @@
+local kong = kong
+local ngx = ngx
+local sub = string.sub
+local split = require('kong.tools.utils').split
+
+local metrics = {}
+
+
+local function init(prometheus)
+  metrics.license_errors = prometheus:counter("enterprise_license_errors",
+                                              "Errors when collecting license info")
+  metrics.license_signature = prometheus:gauge("enterprise_license_signature",
+                                              "Last 32 bytes of the license signautre in number")
+  metrics.license_expiration = prometheus:gauge("enterprise_license_expiration",
+                                                "Unix epoch time when the license expires, " ..
+                                                "the timestamp is substracted by 24 hours "..
+                                                "to avoid difference in timezone")
+  metrics.license_features = prometheus:gauge("enterprise_license_features",
+                                                "License features features",
+                                              { "feature" })
+end
+
+local function license_date_to_unix(yyyy_mm_dd)
+  local date_t = split(yyyy_mm_dd, "-")
+
+  local ok, res = pcall(os.time, {
+    year = tonumber(date_t[1]),
+    month = tonumber(date_t[2]),
+    day = tonumber(date_t[3])
+  })
+  if ok then
+    return res
+  end
+
+  return nil, res
+end
+
+local function metric_data()
+  if not metrics then
+    kong.log.err("prometheus: plugin is not initialized, please make sure ",
+                 " 'prometheus_metrics' shared dict is present in nginx template")
+    return kong.response.exit(500, { message = "An unexpected error occurred" })
+  end
+
+  if not kong.license or not kong.license.license then
+    metrics.license_errors:inc()
+    kong.log.err("cannot read kong.license when collecting license info")
+    return
+  end
+
+  local lic = kong.license.license
+
+  if lic.version ~= 1 then
+    metrics.license_errors:inc()
+    kong.log.err("enterprise license version (" .. (lic.version or "nil") .. ") unsupported")
+    return
```

Comment:
`license_errors` is a counter, but we are bailing out upon encountering the first error; while some errors might be fatal, is it the expected behavior?

Reply:
yes generally we don't want to stop providing any metrics at all if something bad happended during generating them, for example if license is really bad, we should still be providing like latency metrics as usual